### ISSUE TITLE
Exclude dependencies of excluded libraries.

### ISSUE THREAD

**Issue by facundominguez** [tweag/clodl#17]
When removing some library from the closure like in 

```Haskell
  library_closure(
    name = "closure",
    srcs = ...,
    excludes = [
      "libjvm\.so.*",
    ],
  )
```

all the dependencies of `libjvm.so` are retained. If the binary does not require those dependencies transitively through other libraries, they shouldn't be included.

**Comment by facundominguez** [comment]
The most practical option seems to be using a combination of `ldd` to get the paths of all dependencies and  `objdump` to traverse the dependency tree of the binary, skipping the excluded libraries.

Some other options that I've considered follow.
----
Write a program which finds the libraries with `objdump` and follows the dependencies instead of using ldd. `/sbin/ldconf -p` and the `RPATH` provide the list of folders in which to search.

The downside is that we have to keep this tool adjusted to the specifics of each platform were clodl is used. An advantage is that we can possibly build the closure of cross-compiled binaries with this approach.
----
Another option is to use lddtree from pax-utils, to distinguish between direct and transitive dependencies. Unfortunately, lddtree produces several megabytes of output which takes a while to produce and consume.

**Comment by mboes** [comment]
Using `scanelf -qn` (from pax-utils) is an option. It's what `lddtree` uses under the hood.

**Comment by facundominguez** [comment]
Work in progress: https://github.com/tweag/clodl/compare/master...fd/deptree-walk

**Comment by facundominguez** [comment]
Merged in e9f1339.
